planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```
@@ -47,74 +51,64 @@ public Itinerary generateItinerary(List<Event> events) {
+        List<ItineraryItem> items = new ArrayList<>();
+        if (optimized) {
+            items = scheduleOptimizedItinerary(events, START, END);
+        }else {
+            items = scheduleItineraryInOrder(events, START, END);
         }
-        List<ItineraryItem> items = scheduleItineraryInOrder(events, START, END, travelTimeGraph);
+        
         Itinerary itinerary = new Itinerary(items, errorMessage);
         return itinerary;
     }
 
-    // Function that creates a graph with the events as vertices and the travelling
-    // time between the events as edges.
-    // The order of this graph will be the same as the order of the events.
-    // For the MVP, real time traffic is NOT used.
-    // TODO: handle invalid addresses
-    private Duration[][] getEventsDistanceMatrix(List<Event> events) {
-        int numOfNodes = events.size();
-        Duration[][] travelTimeGraph = new Duration[numOfNodes][numOfNodes];
-
-        String GoogleApiKey = "AIzaSyDK36gDoYgOj4AlbCqh1IuaUuTlcpKF0ns";
-        String[] origins = getListOfAddresses(events);
-        String[] destinations = getListOfAddresses(events);
-        GeoApiContext context = new GeoApiContext.Builder(new GaeRequestHandler.Builder())
-                            .apiKey(GoogleApiKey)
-                            .build();
-       try {
-            DistanceMatrix distanceMatrix = DistanceMatrixApi.getDistanceMatrix(context,
-                                                         origins,
-                                                         destinations).await();
+    // Function that creates an itinerary by scheduling each event in order.
+    private List<ItineraryItem> scheduleItineraryInOrder(List<Event> events, int openningTime, int endingTime) {
+        List<ItineraryItem> items = new ArrayList<>();
+        boolean optimized = false;
+        DirectionsRoute directionsRoute = getDirectionsRoute(events, optimized);
+        if (errorMessage != null) { 
```

When can this be true? You are not setting |errorMessage| in this method by this point.

---

errorMessage is global field in the itinerary generator class, so getDirectionRoute fails the errormessage would not be null

---

In what circumstances is errorMessage set? It seems like you could let getDirectionsRoute throw an Exception and catch it here or let it bubble up (which would be more idiomatic).

Alternatively, if you don't want to deal with exceptions, you could return an Optional<DirectionsRoute>: https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html